internal/provider/deployment.go

```diff
@@ -161,10 +161,10 @@ func (d *DeploymentDeployer) GenerateVersionlessDeployments(ctx context.Context)
 		dl.Workloads = append(dl.Workloads, qsfsWorkload)
```

Comment:
on line 153, the function called `vm.GenerateVMWorkload()` while it returns a slice of workloads, better to rename it to `GenerateVMWorkloads`